### PR TITLE
New Progress statuses ahead of new cubiks integration

### PR DESCRIPTION
**Pull request by henricook**

### CODE REVIEW

File: app/model/ApplicationStatuses.scala [hunk 1]
```diff
@@ -23,10 +23,6 @@ object ApplicationStatuses {
   val Withdrawn = "WITHDRAWN"
```
Comment (lukastymo) [inline app/model/ApplicationStatuses.scala]: We can add ```@deprecated``` on the top of this file to show clearly that we shouldn't add new statuses here.

File: app/model/ProgressStatuses.scala [hunk 2]
```diff
@@ -47,4 +47,40 @@ object ProgressStatuses {
...
+
+  case object PHASE1_TESTS_STARTED extends ProgressStatus(ApplicationStatus.PHASE1_TESTS)
+
+  case object PHASE1_TESTS_COMPLETED extends ProgressStatus(ApplicationStatus.PHASE1_TESTS)
+
+  case object PHASE1_TESTS_EXPIRED extends ProgressStatus(ApplicationStatus.PHASE1_TESTS)
+
+  case object PHASE1_TESTS_RESULTS_RECEIVED extends ProgressStatus(ApplicationStatus.PHASE1_TESTS)
+
+  val nameToProgressStatus = List(
```
Comment (lukastymo) [inline app/model/ProgressStatuses.scala]: - add private
- There is some problem with the type (intellij show an error in the line where we use this map). To solve this you can try to add explicit type: ```Map[String, ProgressStatus]```

File: app/model/ProgressStatuses.scala [hunk 3]
```diff
@@ -47,4 +47,40 @@ object ProgressStatuses {
   val AssessmentCentreFailedProgress = ApplicationStatuses.AssessmentCentreFailed.toLowerCase()
   val AssessmentCentrePassedNotifiedProgress = ApplicationStatuses.AssessmentCentrePassedNotified.toLowerCase()
   val AssessmentCentreFailedNotifiedProgress = ApplicationStatuses.AssessmentCentreFailedNotified.toLowerCase()
+
+  sealed abstract class ProgressStatus(applicationStatus: ApplicationStatus)
+
+  object ProgressStatus {
+    implicit val progressStatusFormat = new Format[ProgressStatus] {
+      def reads(json: JsValue) = JsSuccess(nameToProgressStatus(json.as[String]))
+      def writes(myEnum: ProgressStatus) = JsString(myEnum.toString)
```
Comment (lukastymo) [inline app/model/ProgressStatuses.scala]: it's not enum anymore: ```myEnum```

File: app/model/ProgressStatuses.scala (hunk 2 above)
Comment (henricook) [inline app/model/ProgressStatuses.scala]: Done

File: app/model/ProgressStatuses.scala (hunk 3 above)
Comment (henricook) [inline app/model/ProgressStatuses.scala]: Done

File: app/model/ApplicationStatuses.scala (hunk 1 above)
Comment (henricook) [inline app/model/ApplicationStatuses.scala]: Done

File: app/model/ProgressStatuses.scala [hunk 4]
```diff
@@ -47,4 +47,40 @@ object ProgressStatuses {
...
+    }
+
+    implicit object BSONEnumHandler extends BSONHandler[BSONString, ProgressStatus] {
+      def read(doc: BSONString) = nameToProgressStatus(doc.value)
+      def write(stats: ProgressStatus) = BSON.write(stats.toString)
+    }
+  }
+
+  case object PHASE1_TESTS_INVITED extends ProgressStatus(ApplicationStatus.PHASE1_TESTS)
```
Comment (lukastymo) [inline app/model/ProgressStatuses.scala]: We need to make sure that everybody agree with the new capital case convention for progress statuses + we need refactor the old ones

Comment (henricook) [inline app/model/ProgressStatuses.scala]: @ollyw How do you feel about new progress statuses in capitals? If nothing else it helps us easily spot if we're using a "new" progress status or an "old" one

**Pull request merged**